### DISCUSSION TITLE
How to stop carousel cycling?

### DISCUSSION THREAD

**Discussion by tesicg** [bootstrap-vue/bootstrap-vue#6316] [Q&A] [answered]
I've tried this:

```
<template lang="pug">
  b-carousel(
    id='categoryRoulette'
    controls
    no-animation
    ref="myRoulette"
  )
```
    <script lang="ts">
    import Vue from 'vue'

    export default Vue.extend({
      mounted(): void {
        this.$refs.myRoulette.pause()
    },

But, I've got the following error:

```
any
Property 'pause' does not exist on type 'Vue | Element | Vue[] | Element[]'. Property 'pause' does not exist on type 'Vue'.Vetur(2339)

**Comment by Hiws** [comment, marked as answer]
The code is correct.
The error looks to be related to you using TypeScript.

You probably need to cast the ref to `BCarousel`, so that it knows the `pause` method exists.
Here's the type file.
<https://github.com/bootstrap-vue/bootstrap-vue/blob/dev/src/components/carousel/index.d.ts>

Simple demo that shows your code is valid (without typescript)
<https://codepen.io/Hiws/pen/qBaLmNK>

**Comment by tesicg** [comment]
Thank you.

This is working too:

```
<template lang="pug">
b-carousel.d-none.d-sm-block(
  id='categoryRoulette'
  controls
  no-animation
  :interval='0'
)
